mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/building/spi/PojoMapperFactory.java

```diff
@@ -31,7 +32,8 @@ protected PojoMapperFactory(PojoBootstrapIntrospector introspector, boolean impl
 	}
 
 	@Override
-	public final PojoMapper<M> createMapper(BuildContext buildContext, ConfigurationPropertySource propertySource) {
+	public final Mapper<PojoTypeMetadataContributor, M> createMapper(BuildContext buildContext,
```

Comment:
Considering `PojoTypeMetadataContributor` is an impl too, I don't think it solves the issue.

Reply:
Right, this particular change was useless, I removed it. And there was a reference to it in a useless protected method in `PojoMappingContributorImpl`, I removed it too.
But it doesn't matter: the commit moves `PojoMapperFactory` to impl and removes the need for SPIs to refer to the `MapperFactory`, so it does solve the problem.

Reply:
I just rebased and pushed.